src/entry.js

```diff
@@ -85,7 +85,24 @@ class Entry {
    */
   static toMultihash (ipfs, entry) {
     if (!ipfs) throw IpfsNotDefinedError()
-    const data = Buffer.from(JSON.stringify(entry))
+    if (!(entry.id && entry.clock && entry.next && entry.payload && (entry.v === 0 || entry.v > 0))) {
+      throw new Error('Invalid entry format')
+    }
+
+    // Ensure `entry` follows the correct format
+    const e = {
+      hash: null,
+      id: entry.id,
+      payload: entry.payload,
+      next: entry.next,
+      v: entry.v,
+      clock: entry.clock,
+    }
+
+    if (entry.sig) Object.assign(e, { sig: entry.sig })
+    if (entry.key) Object.assign(e, { key: entry.key })
```

Comment:
Instead of converting the data structure to the "right format" here, I feel a better way to handle an invalid entry would be to throw an error if the entry is not valid.

Reply:
Right, this might be the root of my confusion. When `toMultihash` is called by `Entry.create`, it takes an entry with no `.hash` and returns a multihash that is then assigned to the entry later. So if we're calling `toMultihash` passing in an entry with a hash, it'd return a different multihash. Is that the expected behavior or should it only take entry with no `.hash` and return the new multihash?

Reply:
If the latter, using `.isEntry` won't help because it expects the entry to have a hash property – which makes sense if you're checking if it is an entry of the log but wouldn't make sense if you're expecting to get an object to put on IPFS for the first time